backends/freecad/init.py

```diff
@@ -86,44 +84,207 @@ def make_drawing(scale,obj):
 
 
 def list_names(doc):
+    '''
+    BOLTS.list_names(document)
+        lists object Labels and object Names
+        off all Part and Part.Feature document objects
+    '''
     print("Label   Name")
     print("------------")
     for part in doc.findObjects():
-        if isinstance(part,Part.Feature):
+        if isinstance(part, Part.Feature):
             print("%s    %s" % (part.Label, part.Name))
 
 
-def add_name(id,in_params=None):
+# ************************************************************************************************
+# names and standards
+
+def get_name(classid):
+    '''
+    BOLTS.get_name()
+        get SaveClassName from classid (which can be retrieved from blt file)
+    '''
+    for n in get_names():
```

Comment:
I think I had a mechanism to do something like this efficiently, but it has been  long ago that I wrote this code, and I can not try it right now, but I guess something like:

repo.iternames(classid=classid)